lib/internal/process/stdio.js

```diff
@@ -2,7 +2,26 @@
 
 exports.getMainThreadStdio = getMainThreadStdio;
 
-function dummyDestroy(err, cb) { cb(err); }
+function dummyDestroy(err, cb) {
+  // SyncWriteStream does not use the stream
+  // destroy mechanism for some legacy reason.
+  // TODO(mcollina): remove when
+  // https://github.com/nodejs/node/pull/26690 lands.
+  if (typeof cb === 'function') {
+    cb(err);
+  }
+
+  // We need to emit 'close' anyway so that the closing
+  // of the stream is observable. We just make sure we
+  // are not going to do it twice.
+  // The 'close' event is needed so that finished and
+  // pipeline work correctly.
+  if (!this._writableState.emitClose) {
+    process.nextTick(() => {
+      this.emit('close');
+    });
```

Comment:
How come the `nextTick` is required in this case?

Reply:
The previous cb() could emitting an error via nextTick. Close should be emitted afterwards.

Reply:
Are you ok/can this PR land, or would you like to see some different resolution for this?